components/brave_wallet_ui/page/async/wallet_page_async_handler.ts

```diff
@@ -218,6 +225,38 @@ handler.on(WalletPageActions.openWalletSettings.getType(), async (store) => {
   })
 })
 
+handler.on(WalletPageActions.getNFTMetadata.getType(), async (store, payload: BraveWallet.BlockchainToken) => {
+  store.dispatch(WalletPageActions.setIsFetchingNFTMetadata(true))
```

Comment:
`getNFTMetadata` action _always_ results in `setIsFetchingNFTMetadata(true)` action. So it's implied. Therefore, reducer for `getNFTMetadata` can simply set `isFetchingNFTMetadata` to `true`.

This handler also finishes with the `updateNFTMetadata` at which point the reducer can set `isFetchingNFTMetadata` to `false`. In an error state, instead of simply logging, it could dispatch either a new action (e.g. `errorGettingNFTMetadata`) or provide an error state in `updateNFTMetadata`. This would reduce the amount of actions fired by this from 4 to 2, improving performance.

nit: Also actions seem to be mixed between "event"-like names and get/set-like names. Usually for redux, event-like names are preferred. Unless the wallet team has made a different decision? This encourages fewer actions to be fired.